module/zcommon/zfs_prop.c

```diff
@@ -748,6 +748,8 @@ zfs_prop_init(void)
 boolean_t
 zfs_prop_delegatable(zfs_prop_t prop)
 {
+	VERIFY3S(prop, >=, 0);
+	VERIFY3S(prop, <, ZFS_NUM_PROPS);
```

Comment:
Using an ASSERT3S instead of VERIFY3S seems like it should be sufficient to ensure we catch any potential future regressions here during testing.

Reply:
Corrected in https://github.com/openzfs/zfs/pull/14147

Reply:
Merged https://github.com/openzfs/zfs/pull/14147